File: sdks/go/identity.go
package openwallet

import (
	"fmt"

	"github.com/google/uuid"
)

type SignMessageResponse struct {
	Signature string `json:"signature"`
	Address   string `json:"address"`
}

type VerifySignatureResponse struct {
	Valid bool `json:"valid"`
}

type SignTransactionResponse struct {
	SignedTx string `json:"signed_tx"`
	TxHash   string `json:"tx_hash"`
}

type IdentityResource struct {
	client *Client
}

func (r *IdentityResource) SignMessage(walletID uuid.UUID, message string) (*SignMessageResponse, error) {
	var resp SignMessageResponse
	err := r.client.doJSON("POST", "/v1/identity/sign-message", map[string]interface{}{
		"wallet_id": walletID,
		"message":   message,
	}, &resp)
	return &resp, err
}

func (r *IdentityResource) VerifySignature(message, signature, address string) (*VerifySignatureResponse, error) {
	var resp VerifySignatureResponse
	err := r.client.doJSON("POST", "/v1/identity/verify-signature", map[string]interface{}{
		"message":   message,
		"signature": signature,
		"address":   address,
	}, &resp)
	return &resp, err
}

func (r *IdentityResource) SignTransaction(walletID uuid.UUID, network, to, value, data string) (*SignTransactionResponse, error) {
	var resp SignTransactionResponse
	err := r.client.doJSON("POST", "/v1/identity/sign-transaction", map[string]interface{}{
		"wallet_id": walletID,
		"network":   network,
		"to":        to,
		"value":     value,
		"data":      data,
	}, &resp)
	return &resp, err
}

File: sdks/typescript/packages/sdk/src/resources/identity.ts
import { HttpClient } from '../client';
import { SignMessageResponse, VerifySignatureResponse } from '../types';

export class IdentityResource {
  constructor(private client: HttpClient) {}

  async signMessage(
    walletId: string,
    message: string,
  ): Promise<SignMessageResponse> {
    return this.client.post<SignMessageResponse>('/v1/identity/sign-message', {
      wallet_id: walletId,
      message,
    });
  }

  async verifySignature(
    message: string,
    signature: string,
    address: string,
  ): Promise<VerifySignatureResponse> {
    return this.client.post<VerifySignatureResponse>(
      '/v1/identity/verify-signature',
      { message, signature, address },
    );
  }

  async signTransaction(
    walletId: string,
    network: string,
    to: string,
    value?: string,
    data?: string,
  ): Promise<{ signed_tx: string; tx_hash: string }> {
    return this.client.post('/v1/identity/sign-transaction', {
      wallet_id: walletId,
      network,
      to,
      value,
      data,
    });
  }
}
